Fix forced backlogged not being limited by BACKLOG_DAYS

fernandog opened the pull request:
as @ratoaq2 pointed

dividing integers will result in integer so
`self.cycleTime = app.BACKLOG_FREQUENCY / 60 / 24` will result 0.5 but value will be 0

so a backlog of 12 hours, will always have a cycleTime of  0 in case of the default of 720

So when checking if we should run a forced search, it will never reach that limit code:
`if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):`
as `(curDate - self._last_backlog)` will always be higher than cycleTime

Imo this code was done to check if user runned a backlog before waiting for next backlog (so, a forced backlog)

![image](https://cloud.githubusercontent.com/assets/2620870/22883321/6dce1ae2-f1ce-11e6-9a13-ad62613771d2.png)

duramato commented:
What is the current behaviour?
If i'm reading correctly, now it will be: episodes with WANTED status older than "Backlog Days" won't ever get searched without user interaction?

fernandog commented:
How ep become wanted? How long is wanted?

duramato commented:
Became wanted becouse it aired, let's say it has been WANTED for longer than the "backlog days setting" value, what is the current behaviour and what will be the behaviour with this PR?

fernandog commented:
I changed this PR and now it only fixed issue described in PR body:

Also added a note in the UI:
![image](https://cloud.githubusercontent.com/assets/2620870/22884723/2da0d804-f1d5-11e6-8b49-777f3eb28aaf.png)

medariox commented on medusa/search/backlog.py:
```diff
@@ -49,6 +49,7 @@ def __init__(self):
         self.amActive = False
         self.amPaused = False
         self.amWaiting = False
+        self.force = False
```
Why do you make a new attribute for this? Just pass it to self.search_backlog(force)

fernandog replied:
because force is also a attribute of the backlog.

medariox replied:
What do you mean?

labrys replied:
If you mean `force` as an attribute (meaning this backlog was triggered as the result of a "forced backlog" action) then the attribute name should be called `forced`.  But as @medariox asked: What do you mean?

fernandog replied:
@labrys Yes. Forced because user clicked the force button on manage searches. I also added a note there

fernandog replied:
@medariox can you accept changes please?

medariox commented on medusa/search/backlog.py:
```diff
@@ -90,9 +91,11 @@ def search_backlog(self, which_shows=None):
-        if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):
-            logger.log(u"Running limited backlog on missed episodes " + str(app.BACKLOG_DAYS) + " day(s) and older only")
+        if not which_shows and self.force:
+            logger.log(u"Running limited backlog search on missed episodes from last {0} days".format(app.BACKLOG_DAYS))
```
Use single quotes

p0psicles commented:
If I run a python interpreter and run 4 / 0.5 / 3 it results in 2.666666666...?

labrys commented:
@p0psicles you have a float in the division thus the result is a float.  Same if you did any of the following:
```python
>>> float(3)/4
0.75

>>> 3.0/4
0.75

>>> 3/4.0
0.75

>>> from  __future__ import division
>>> 3/4  # after future import
0.75
>>> 3 // 4  # for integer division after future import
0
```

p0psicles commented:
aah yeah, now I see what he meant. well then if you expect a float, just divide by 24.0 right?

labrys commented:
You should actually promote first element to avoid this kind of issue:
```python
>>> 3 / 4 / 0.5
0

>>> 3.0 / 4 / 0.5
1.5
```

labrys commented:
Personally I prefer to use `__future__ division` everywhere and use `//` if I need explicit integer division

labrys commented:
Regardless integer division isnt actually a bug here.
It's being used at L93 in backlog.py as follows:
```
        if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):
```
And curDate is a date (min resolution is a day) so 0 is the correct answer for a backlog frequency.

fernandog commented:
@labrys the forced is better than do all that math right? Its what the feature is called "forced search" will use backlog days

fernandog commented:
@labrys about frequency, in server status it show cycle time of 12 hours

fernandog commented:
@labrys changed to `forced` and promoted to float. btw he backlog_days limit doesn't use that cycleTime anymore

fernandog commented:
Another PR ready waiting for approval
